gemhardware/devices/Makefile

```diff
@@ -57,7 +57,7 @@ LibraryDirs+=$(BUILD_HOME)/gemutils/lib/$(XDAQ_OS)/$(XDAQ_PLATFORM)
 LibraryDirs+=$(uHALROOT)/lib
 LibraryDirs+=$(XHAL_ROOT)/lib
 
-DependentLibraries =cactus_uhal_uhal xhal
+DependentLibraries =cactus_uhal_uhal xhal-client
```

Comment:
does it not also depend on the `xhal-base` library?

Reply:
What do you mean by "it"? The PR or the `xhal-client` library? If you are referring to library, the implicit dependency shouldn't be added here (this is not a static library). If you are referring to the PR, yes. However the `common` namespace is misleading with respect to the library `-base` suffix.

Reply:
> What do you mean by "it"? The PR or the `xhal-client` library? If you are referring to library, the implicit dependency shouldn't be added here (this is not a static library). If you are referring to the PR, yes. However the `common` namespace is misleading with respect to the library `-base` suffix.

"it" == `libgemhardware.so`

also, the rpm package is simply `xhal` :-D (though it was mentioned that this could be a dummy package and the actual `xhal-base` library and friends could be packaged as `xhal-base`...

Reply:
actually I don't think it should depend on base - why? It doesn't actually use `base` functionality, it uses the `client` one. Moreover, `base` only goes to `arm` in *glibless* world

Reply:
> actually I don't think it should depend on base - why? It doesn't actually use `base` functionality, it uses the `client` one. Moreover, `base` only goes to `arm` in _glibless_ world

`base` provides the core RPC translator functionality, so it winds up on both ends of the connection

Reply:
_If_ you don't explicitly use `base` classes or functions in the code, you shouldn't have to depend on it (or else the build system should be considered as broken).

In this PR, classes inside `xhal::common::utils` _are_ explicitly used; nothing guarantees that `xhal-client` will _always_ depends on `xhal` (without `-base` :-D)